shared/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -155,6 +155,82 @@ namespace datadog::shared::nativeloader
             }
         }
 
+        //
+        // Get Profiler interface ICorProfilerInfo4
+        //
+        ICorProfilerInfo4* info4 = nullptr;
+        HRESULT hr = pICorProfilerInfoUnk->QueryInterface(__uuidof(ICorProfilerInfo4), (void**) &info4);
+        if (FAILED(hr))
+        {
+            Log::Warn("CorProfiler::Initialize: Failed to attach profiler, interface ICorProfilerInfo4 not found.");
+            return E_FAIL;
+        }
+        const auto runtimeInformation = GetRuntimeVersion(info4);
+
+        //
+        // Check if we're running in Single Step, and if so, whether we should bail out
+        //
+        // This variable is non-empty when we're in single step
+        const auto isSingleStepVariable = GetEnvironmentValue(EnvironmentVariables::SingleStepInstrumentationEnabled);
+        if(!isSingleStepVariable.empty())
```

Comment:
so even `DD_INJECTION_ENABLED=false` works? 😅

Reply:
Yes, it would, but this is an internal variable that is set by SSI and currently is only set to `tracer` or `service-name`. I don't believe there's any intention to inject booleans

Reply:
Ok, just making sure you considered this possibility. Thanks!